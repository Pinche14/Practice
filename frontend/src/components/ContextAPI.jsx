import React, {useEffect, useState} from 'react'
import PostContainer from './PostContainer.jsx/PostContainer'
import { UserCOntext } from '../utlis/contexts/UserCOntext'
import DataFetching from '../utlis/hooks/DataFetching'
import { Link, Outlet, useNavigate } from 'react-router-dom'
const ContextAPI = () => {
  const {user, loading, error} = DataFetching(7)
  console.log(user,loading,error)

  const [userData, setUserData] = useState(
  //   id:1,
  //   username: "Shubham Pathak",
  //   email: "shubham123@gmail.com",
  //   displayName: "Charlatan"
  )

  const navigate = useNavigate()
  useEffect(() => {
    if(!loading && !error && user) {
      setUserData(user)
      navigate('/users')
    }
  }, [loading,error,user,navigate]);
  return (
    <>
      <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/users">Users</Link>
          </li>
          <li>
            <Link to="/blog-posts">BlogPosts</Link>
          </li>
        </ul>
      </nav>

      <div>
        <label htmlFor="data">Enter Data</label>
        <input type="text" id="data" onChange={(e.atget.value)}/>
      </div>
      <UserCOntext.Provider value={{...userData, setUserData}}>
      <div>
        {loading ? 'loading...': <PostContainer/>}
      </div>
      </UserCOntext.Provider>
      <Outlet/>
    </>
    
    // <div>
    //   <PostContainer/>
    // </div>
  )
}

export default ContextAPI
