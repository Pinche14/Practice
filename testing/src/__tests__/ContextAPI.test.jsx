import {expect, it, describe} from 'vitest'
import {render,screen} from "@testing-library/react"
import ContextAPI from '../components/ContextAPI'
import userEvent from '@testing-library/user-event'

describe('ContextAPI', () => {
    it('should render users', () => {
        const {container} = render(<ContextAPI/>);
        expect(container).toMatchSnapshot();
    })


    describe('Edit button' , () => {
        it('should render save button',async () => {
            render(<ContextAPI/>)
            const editButton = screen.getByRole('button', {name: 'Edit'});
            await userEvent.click(editButton)
            const saveButton = screen.getByRole('button', {name: "Save"});
            expect(saveButton).toBeInTheDocument();
        })
        it('should display username & email input fields when button is clicked',async () => {
            render(<ContextAPI/>)
            const render = screen.getByRole('button', {name: "Edit"});
            await userEvent.click(editButton)
        })
    
    })
})