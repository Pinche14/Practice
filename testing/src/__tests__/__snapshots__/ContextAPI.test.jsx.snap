// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextAPI > Edit button > should render users 1`] = `
<div>
  <div>
    <div>
      <button>
        Edit
      </button>
      <button>
        Delete
      </button>
    </div>
    <div>
      <b>
        ID:
        1
         
      </b>
      <br />
      <b>
        Username: 
      </b>
      <span>
        Shubham
      </span>
      <br />
      <b>
        Email: 
      </b>
      <span>
        shubham123@gmail.com
      </span>
    </div>
  </div>
</div>
`;

exports[`ContextAPI > should render users 1`] = `
<div>
  <div>
    <div>
      <button>
        Edit
      </button>
      <button>
        Delete
      </button>
    </div>
    <div>
      <b>
        ID:
        1
         
      </b>
      <br />
      <b>
        Username: 
      </b>
      <span>
        Shubham
      </span>
      <br />
      <b>
        Email: 
      </b>
      <span>
        shubham123@gmail.com
      </span>
    </div>
  </div>
</div>
`;
